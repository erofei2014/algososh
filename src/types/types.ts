import { Direction } from "./direction";

export type TCircleStatus = {
  index: number;
  steps: string[][];
  currStepIndex: number;
};

export type TColumnStatus = {
  index: number;
  steps: [number[], number, number, number[]][];
  currStepIndex: number;
};

export type TSort = {
  active: boolean;
  direction: Direction;
};

export type TStackStatus = {
  index: number;
  lastIndex: number;
  currStepIndex: number;
}

export type TQueueStatus = {
  index: number;
  keyIndex: number;
  currStepIndex: number;
}