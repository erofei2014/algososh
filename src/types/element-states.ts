export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}

export enum ColorTest {
  Default = '4px solid rgb(0, 50, 255)',
  Changing = '4px solid rgb(210, 82, 225)',
  Modified = '4px solid rgb(127, 224, 81)',
}
