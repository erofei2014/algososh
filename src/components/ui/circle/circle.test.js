import React from 'react';
import { Circle } from './circle';
import renderer from 'react-test-renderer';
import { ElementStates } from '../../../types/element-states';

describe("Компонент Circle", () => {
  it("Отрисовка пустого круга", () => {
    const tree = renderer
    .create(<Circle />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка круга с текстом", () => {
    const tree = renderer
    .create(<Circle letter={'12'}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка круга с head", () => {
    const tree = renderer
    .create(<Circle head={'head'}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка круга с react-элементом в head", () => {
    const reactElement = <div></div>
    const tree = renderer
    .create(<Circle head={reactElement}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка круга с tail", () => {
    const tree = renderer
    .create(<Circle tail={'tail'}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка круга с react-элементом в tail", () => {
    const reactElement = <div></div>
    const tree = renderer
    .create(<Circle tail={reactElement}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка круга с индексом", () => {
    const tree = renderer
    .create(<Circle index={1}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка малого круга", () => {
    const tree = renderer
    .create(<Circle isSmall={true}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка круга с дефолтной рамкой", () => {
    const tree = renderer
    .create(<Circle state={ElementStates.Default} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("Отрисовка круга с красной рамкой", () => {
    const tree = renderer
    .create(<Circle state={ElementStates.Changing} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка круга с зеленой рамкой", () => {
    const tree = renderer
    .create(<Circle state={ElementStates.Modified} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});