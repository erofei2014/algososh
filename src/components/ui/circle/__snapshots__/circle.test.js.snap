// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Circle Отрисовка круга с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div />
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div />
  </div>
</div>
`;

exports[`Компонент Circle Отрисовка круга с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Компонент Circle Отрисовка круга с дефолтной рамкой 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга с зеленой рамкой 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга с индексом 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга с красной рамкой 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга с текстом 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      12
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка малого круга 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка пустого круга 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
